runtime/gateway.go

```diff
@@ -434,6 +434,9 @@ func (gateway *Gateway) Shutdown() {
 		}
 	}()
 
+	// wait for servers to shutdown before stopping GRPCClientDispatcher
+	swg.Wait()
```

Comment:
with this change, you can make only 3 channels on line# 407, make the following gRPC shutdown run sequentially and remove the finally wait on line# 451.